### ISSUE TITLE
[Core feature] flyekit should ship PEP-561 compatible stubs (support type linters e.g. mypy)

### ISSUE THREAD

**Issue by asford** [flyteorg/flyte#2147]
### Motivation: Why do you think this is important?

`flytekit` already has relatively extensive internal type annotations, and is check by `mypy` as a linting pass for the library.
The distribution package should include a [PEP 561] compatible `py.typed` entry, so that consuming applications can access and use this type annotations.

This should be a very minimal change to the existing distribution process.

[PEP 561]: https://www.python.org/dev/peps/pep-0561/#packaging-type-information

### Goal: What should the final outcome look like, ideally?

Add `py.typed` to the `flytekit` distribution, so that `mypy` runs in applications consuming `flytekit` can consume type annotations during standard check runs.

### Describe alternatives you've considered 

We could generate type stubs via `mypy` and include in applications or in a stub package, however this introduces drift against the `flykite` due to version changes.

### Are you sure this issue hasn't been raised already?

- [X] Yes

### Have you read the Code of Conduct?

- [X] Yes

**Comment by pingsutw** [comment]
cc @eapolinario @wild-endeavor should we add Add `py.typed`?

**Comment by kumare3** [comment]
> cc @eapolinario @wild-endeavor should we add Add `py.typed`?

Yes we should, but we have to fix more. But let's wait for March
